sentry-spring/src/main/java/io/sentry/spring/SentryUserProviderEventProcessor.java

```diff
@@ -0,0 +1,48 @@
+package io.sentry.spring;
+
+import io.sentry.core.EventProcessor;
+import io.sentry.core.SentryEvent;
+import io.sentry.core.protocol.User;
+import io.sentry.core.util.Objects;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class SentryUserProviderEventProcessor implements EventProcessor {
+  private final @NotNull SentryUserProvider sentryUserProvider;
+
+  public SentryUserProviderEventProcessor(final @NotNull SentryUserProvider sentryUserProvider) {
+    this.sentryUserProvider =
+        Objects.requireNonNull(sentryUserProvider, "sentryUserProvider is required");
+  }
+
+  @Override
+  public SentryEvent process(final @NotNull SentryEvent event, final @Nullable Object hint) {
+    final User user = sentryUserProvider.provideUser();
+    if (user != null) {
+      final User existingUser = Optional.ofNullable(event.getUser()).orElseGet(User::new);
+
+      Optional.ofNullable(user.getEmail()).ifPresent(existingUser::setEmail);
+      Optional.ofNullable(user.getId()).ifPresent(existingUser::setId);
+      Optional.ofNullable(user.getIpAddress()).ifPresent(existingUser::setIpAddress);
+      Optional.ofNullable(user.getUsername()).ifPresent(existingUser::setUsername);
+      if (user.getOthers() != null && !user.getOthers().isEmpty()) {
+        if (existingUser.getOthers() == null) {
+          existingUser.setOthers(new ConcurrentHashMap<>());
+        }
+        for (Map.Entry<String, String> entry : user.getOthers().entrySet()) {
+          existingUser.getOthers().put(entry.getKey(), entry.getValue());
+        }
+      }
```

Comment:
does it make sense to assign the `getOthers` to a variable instead of calling the getters multiple times? I remember @bruno-garcia advising me to do this multiple times.

Reply:
You got to convince me that this is just the Java way, that these getters are _expected to be cheap_. I'm happy to back off.

My reasoning comes from C# that properties (that read like field) even though are methods are not expected to have side-effect or to be costly to be called.
Methods on the other hand the opposite. Because of that when I see a call to a method I immediately think of assigning the result to use the same reference.

It's also relevant for instances accessed by multiple thread, so you get a consistent view of some state by inspecting the same instance (say it's reading-only operations) while another thread could swap out the reference the getter returns in between the reads.

Since I'm the odd ball here, you can ignore me and go ahead with the `getX()` as it's the "Java way".

Reply:
you forgot to update me on that though haha, thanks for explaining!